### PR TITLE
add checkParams() function

### LINKED ISSUE #245: Do not accept params that we don’t know how to process
This should catch errors such as using `--tool` instead of the correct `--tools`.

### PR DESCRIPTION
**Pull request by MaxUlysse**
fixes #245

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -1157,6 +1161,32 @@ def checkParameterList(list, realList) {
   return list.every{ checkParameterExistence(it, realList) }
 }
 
+def checkParams(it) {
+ // Check if params is in this given list
+  return it in [
+  'callName',
+  'call-name',
```
Comment (marcelm) [inline main.nf]: What’s the difference between these (`callName` and `call-name`)? Do we need both?

File: main.nf [hunk 2]
```diff
@@ -56,6 +56,10 @@ vim: syntax=groovy
 
 version = '1.1'
 
+params.each{
+  if (!checkParams(it.toString().split('=')[0])) {exit 1, "params $it.toString().split('=')[0] is unknown, see --help for more information"}
```
Comment (marcelm) [inline main.nf]: Can’t you move the full code into `checkParams`? As it is, `checkParams` really is `isAllowedParam`.

**Comment by MaxUlysse** [comment]
Make senses, I'm on it ;-)

**Comment by MaxUlysse** [comment]
@szilvajuhos You're a little too quick to merge PR...

File: main.nf (hunk 1 above)
Comment (MaxUlysse) [inline main.nf]: We do need both.
When I loop into params, to see every parameters, whenever there is a parameter with a uppercase letter, it adds another parameters with a - and a lower case letter to replace the first uppercase letter.
There is probably something to explain that, or maybe I didn't access it the right way...

**Pull request merged**